File: 0202-happy-number/0202-happy-number.cpp
int helper(int num) {
    int sum = 0;
    while (num > 0) {
        int d = num % 10;
        num = num / 10;
        sum = sum + d * d;
    }
    return sum;
}

class Solution {
public:
    bool isHappy(int n) {
        int slow = n;
        int fast = n;
        while (fast != 1) {
            slow = helper(slow);
            fast = helper(fast);
            fast = helper(fast);
            if (slow == fast && slow != 1) return false;
        }
        return true;
    }
};

File: 0202-happy-number/0202-happy-number.go
func isHappy(n int) bool {
    slow := n;
    fast := n;
    for fast != 1 {
        slow = helper(slow);
        fast = helper(fast);
        fast = helper(fast);
        if slow == fast && slow != 1 {
            return false;
        }
    }
    return true;
}

func helper(num int) int {
    sum := 0;
    for num > 0 {
        d := num % 10;
        num = num / 10;
        sum = sum + d * d;
    }
    return sum;
}